admin-dev/themes/new-theme/js/pages/product/edit/combinations-grid-renderer.ts

```diff
@@ -118,7 +123,7 @@ export default class CombinationsGridRenderer {
    *
    * @private
    */
-  getPrototypeRow(rowIndex) {
-    return this.prototypeTemplate.replace(new RegExp(this.prototypeName, 'g'), rowIndex);
+  private getPrototypeRow(rowIndex: number): string {
+    return this.prototypeTemplate.replace(new RegExp(this.prototypeName, 'g'), rowIndex.toString());
```

Comment:
```suggestion
    return this.prototypeTemplate.replace(new RegExp(this.prototypeName, 'g'), <string>rowIndex);
```
Isn't it equivalent?

Reply:
- Casting an int into a string is not natural at all, this is why you need to cast it to unknown before
- replace takes a string as a parameter, it's on language level, not on typescript level, we need to use toString

Reply:
Oh so `<string>` is only to make TypeScript happy but it doesn't actually cast the value? So if it works it means that it's javascript that performs auto casting I guess?